packages/collector/Collector_Db.php

```diff
@@ -0,0 +1,82 @@
+<?php
+function collector_dump_db($zip)
+{
+	$tables   = collector_get_db_tables();
+	$sqlFile  = collector_get_tmpfile('schema', 'sql');
+	$tmpFiles = [$sqlFile];
+
+	file_put_contents($sqlFile, sprintf("-- %s\n", json_encode(['SECTION START' => 'SCHEMA'])), FILE_APPEND);
+
+	foreach($tables as $table)
+	{
+		file_put_contents($sqlFile, sprintf("-- %s\n", json_encode(['ACTION' => 'DROP', 'TABLE' => $table])), FILE_APPEND);
+		file_put_contents($sqlFile, sprintf("DROP TABLE IF EXISTS `%s`;\n", $table), FILE_APPEND);
+		file_put_contents($sqlFile, sprintf("-- %s\n", json_encode(['ACTION' => 'CREATE', 'TABLE' => $table])), FILE_APPEND);
+		file_put_contents($sqlFile, preg_replace("/\s+/", " ", collector_dump_db_schema($table)) . "\n", FILE_APPEND);
+	}
+
+	file_put_contents($sqlFile, sprintf("-- %s\n", json_encode(['SECTION END' => 'SCHEMA'])), FILE_APPEND);
+	file_put_contents($sqlFile, sprintf("-- %s\n", json_encode(['SECTION START' => 'RECORDS'])), FILE_APPEND);
+
+	global $wpdb;
+
+	// Process in reverse order so wp_users comes before wp_options
+	// meaning the fakepass will be cleared before transients are
+	// dumped to the schema backup in the zip
+	foreach(array_reverse($tables) as $table)
+	{
+		file_put_contents($sqlFile, sprintf("-- %s\n", json_encode(['ACTION' => 'INSERT', 'TABLE' => $table])), FILE_APPEND);
+
+		$wpdb->query(sprintf('SELECT * FROM `%s`', esc_sql($table)));
+
+		$remaining = $wpdb->result->num_rows;
+
+		if(!$remaining)
+		{
+			continue;
+		}
+
+		foreach($wpdb->result as $record)
+		{
+			if($table === 'wp_users' && (int) $record['ID'] === (int) wp_get_current_user()->ID)
+			{
+				$record['user_pass'] = wp_hash_password(collector_use_fakepass());
+			}
+
+			$insert = sprintf(
+				'INSERT INTO `%s` (%s) VALUES (%s);',
+				esc_sql($table),
+				implode(', ', array_map(fn($f) => "`" . esc_sql($f) . "`", array_keys($record))),
```

Comment:
Does `esc_sql` escape backticks?

Reply:
TIL you can have backticks in an sql identifier. Turns out you escape them by doubling, not with backslashes.

https://github.com/WordPress/playground-tools/pull/124/files#diff-a9e52be37ae14ddda1c496c7a7f66a2139441a47bf645aab6ebb0e798cdb2ba3R54